operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -320,22 +327,29 @@ func (r *TenantReconciler) createOrUpdateClusterResources(ctx context.Context, t
 	}
 
 	var retErr error
+	// retrieve instances
+	instances := crownlabsv1alpha2.InstanceList{}
+	err = r.Client.List(ctx, &instances, &client.ListOptions{Namespace: nsName})
+	if err != nil {
+		klog.Errorf("Unable to retrieve instances")
+		retErr = err
+	}
 	// handle resource quota
 	rq := v1.ResourceQuota{
 		ObjectMeta: metav1.ObjectMeta{Name: "crownlabs-resource-quota", Namespace: nsName},
 	}
 	rqOpRes, err := ctrl.CreateOrUpdate(ctx, r.Client, &rq, func() error {
 		rq.Labels = r.updateTnResourceCommonLabels(rq.Labels)
-		rq.Spec.Hard = forge.TenantResourceQuotaSpec(&tn.Status.Quota)
-
+		rq.Spec.Hard = forge.TenantResourceQuotaSpec(tn.Status.Quota.Limits)
+		tn.Status.Quota.Used = forge.TenantResourceQuotaStatusUsed(&rq, &instances)
```

Comment:
This is the resource quota mutation function, and should be used to mutate the resource quota, and not extract additional values. That can be done as a subsequent step accessing the `rq` value.